### PR TITLE
removed `register_plugin` construct

### PR DESCRIPTION
**Pull request by jstucke**
Removed `register_plugin` and `plugin_administrator` constructs where the scheduler class is passed to the plugin so that the plugin can write itself into the list of plugins inside the scheduler, which is both unnecessarily complicated and generally unnecessary.
Instead, this can be done directly inside the scheduler (where the plugins are initialized anyway).

### CODE REVIEW

File: src/config/main.cfg [hunk 1]
```diff
@@ -79,6 +79,9 @@ threads = 2
 [crypto_material]
 threads = 2
 
+[cve_lookup]
+threads = 3
```
Comment (maringuu) [inline src/config/main.cfg:83]: Were the changes in this file included by accident?

If not please put them in a seperate commit.

File: src/scheduler/analysis.py [hunk 2]
```diff
@@ -418,7 +407,7 @@ def _start_result_collector(self):
         self.result_collector_process = ExceptionSafeProcess(target=self._result_collector)
         self.result_collector_process.start()
 
-    def _result_collector(self):  # pylint: disable=too-complex
```
Comment (maringuu) [inline src/scheduler/analysis.py:421]: Nit: This change should also be a separate commit.

**Review by maringuu — CHANGES_REQUESTED** [review]
Good to finally have this thing removed.
Just the two comments and this should be good to go.

File: src/config/main.cfg (hunk 1 above)
Comment (jstucke) [reply src/config/main.cfg]: That was not intentional -> removed

File: src/scheduler/analysis.py (hunk 2 above)
Comment (jstucke) [reply src/scheduler/analysis.py:421]: I removed it because it caused this:

> src/scheduler/analysis.py:410: [bad-option-value:] Bad option value 'too-complex'

I'm not even sure why. Was this option removed in a recent pylint version?

Comment (maringuu) [reply src/scheduler/analysis.py:421]: Hmm I dont get this error.

```
$ pylint --version
pylint 2.15.3
astroid 2.12.10
Python 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
```

**Review by maringuu — APPROVED** [review]
